File: fundamentos/desafio-js/src/questions/total-higher-than-100.ts
import { IAccount } from "../i-account";

/** O número total de contas com mais de 100 reais de saldo é: */
export default function totalHigherThan100(...accounts: Array<IAccount>) {
  const higherThan100 = accounts
    .filter(acc => acc.balance > 100)
    .length;
  console.info(`O número total de contas com mais de 100 reais de saldo é: ${higherThan100}`);
}

File: fundamentos/desafio-dart/lib/questions/total_higher_than_100.dart
import 'package:desafio_dart/domains/account.dart';

/// O número total de contas com mais de 100 reais de saldo é:
void totalHigherThan100(List<Account> accounts) {
  final higherThan100 = accounts.where((acc) => acc.balance > 100).length;

  print(
      'O número total de contas com mais de 100 reais de saldo é: ${higherThan100}');
}
